libcontainer/cgroups/utils.go

```diff
@@ -137,21 +137,30 @@ func GetCgroupMounts() ([]Mount, error) {
 		allMap[s] = true
 	}
 
-	res := []Mount{}
-	for _, mount := range mounts {
-		if mount.Fstype == "cgroup" {
-			m := Mount{Mountpoint: mount.Mountpoint, Root: mount.Root}
-
-			for _, opt := range strings.Split(mount.VfsOpts, ",") {
-				if strings.HasPrefix(opt, cgroupNamePrefix) {
-					m.Subsystems = append(m.Subsystems, opt[len(cgroupNamePrefix):])
-				}
-				if allMap[opt] {
-					m.Subsystems = append(m.Subsystems, opt)
-				}
+	var res []Mount
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		txt := scanner.Text()
+		fields := strings.Split(txt, " ")
+		if fields[len(fields)-3] != "cgroup" {
```

Comment:
I meant that we can look for the separator -and then match the first word after instead of going minus 3 :)  discretionary though :)

Sent from my iPhone

> On Jan 30, 2016, at 4:43 AM, Alexander Morozov <notifications@github.com> wrote:
> 
> In libcontainer/cgroups/utils.go:
> 
> > -		if mount.Fstype == "cgroup" {
> > -			m := Mount{Mountpoint: mount.Mountpoint, Root: mount.Root}
> > -
> > -			for _, opt := range strings.Split(mount.VfsOpts, ",") {
> > -				if strings.HasPrefix(opt, cgroupNamePrefix) {
> > -					m.Subsystems = append(m.Subsystems, opt[len(cgroupNamePrefix):])
> > -				}
> > -				if allMap[opt] {
> > -					m.Subsystems = append(m.Subsystems, opt)
> > -				}
> > +	var res []Mount
> > +	scanner := bufio.NewScanner(f)
> > +	for scanner.Scan() {
> > +		txt := scanner.Text()
> > +		fields := strings.Split(txt, " ")
> > +		if fields[len(fields)-3] != "cgroup" {
> @mrunalp what do you mean? :)
> 
> —
> Reply to this email directly or view it on GitHub.
>